cmd/dp-hierarchy-builder/main.go

```diff
@@ -2,127 +2,194 @@ package main
-	log.Debug("loaded config", log.Data{"cfg": cfg})
+	log.Event(ctx, "loaded config", log.INFO, log.Data{"cfg": cfg})
 
 	// a channel used to signal a graceful exit is required.
 	errorChannel := make(chan error)
 
 	kafkaBrokers := cfg.KafkaAddr
+	cgChannels := kafka.CreateConsumerGroupChannels(true)
 	kafkaConsumer, err := kafka.NewConsumerGroup(
+		ctx,
 		kafkaBrokers,
 		cfg.ConsumerTopic,
 		cfg.ConsumerGroup,
 		kafka.OffsetNewest,
+		true,
+		cgChannels,
 	)
-	exitIfError(err)
+	exitIfError(ctx, err, "error creating kafka consumer")
 
-	kafkaProducer, err := kafka.NewProducer(kafkaBrokers, cfg.ProducerTopic, 0)
-	exitIfError(err)
+	pChannels := kafka.CreateProducerChannels()
+	useDefaultMaxMessageSize := 0 // pass zero to use the default
+	kafkaProducer, err := kafka.NewProducer(ctx, kafkaBrokers, cfg.ProducerTopic, useDefaultMaxMessageSize, pChannels)
+	if err != nil {
+		log.Event(ctx, "error creating kafka producer", log.FATAL, log.Error(err))
+		os.Exit(1)
+	}
 
-	kafkaErrorProducer, err := kafka.NewProducer(cfg.KafkaAddr, cfg.ErrorProducerTopic, 0)
-	exitIfError(err)
+	errorProducerChannels := kafka.CreateProducerChannels()
+	kafkaErrorProducer, err := kafka.NewProducer(ctx, kafkaBrokers, cfg.ErrorProducerTopic, useDefaultMaxMessageSize, errorProducerChannels)
+	exitIfError(ctx, err, "error creating kafka producer")
 
 	avroProducer := event.NewAvroProducer(kafkaProducer)
 
-	db, err := graph.NewHierarchyStore(context.Background())
-	exitIfError(err)
+	db, err := graph.NewHierarchyStore(ctx)
+	exitIfError(ctx, err, "error creating Kafka error producer")
 
 	// when errors occur - we send a message on an error topic.
 	errorHandler, err := reporter.NewImportErrorReporter(kafkaErrorProducer, log.Namespace)
-	exitIfError(err)
+	exitIfError(ctx, err, "error creating import error reporter")
 
 	eventHandler := event.NewDataImportCompleteHandler(&hierarchy.Store{db}, avroProducer)
 
 	eventConsumer := event.NewConsumer()
-	eventConsumer.Consume(kafkaConsumer, eventHandler, errorHandler)
-
-	healthChecker := healthcheck.NewServer(
-		cfg.BindAddr,
-		cfg.HealthCheckInterval,
-		cfg.HealthCheckRecoveryInterval,
-		errorChannel,
-		db,
-	)
+	eventConsumer.Consume(ctx, kafkaConsumer, eventHandler, errorHandler)
+
+	hc := startHealthChecks(ctx, cfg, kafkaConsumer, kafkaProducer, kafkaErrorProducer, db)
+
+	apiErrors, httpServer := startApi(ctx, hc, cfg)
 
 	signals := make(chan os.Signal, 1)
 	signal.Notify(signals, os.Interrupt, syscall.SIGTERM)
 
 	// this will block (main) until a fatal error occurs
 	select {
-	case err := <-kafkaConsumer.Errors():
-		log.ErrorC("kafka consumer", err, nil)
-	case err := <-kafkaProducer.Errors():
-		log.ErrorC("kafka result producer", err, nil)
-	case err := <-kafkaErrorProducer.Errors():
-		log.ErrorC("kafka error producer", err, nil)
+	case err := <-apiErrors:
+		log.Event(ctx, "http server error", log.ERROR, log.Error(err))
+	case err := <-kafkaConsumer.Channels().Errors:
```

Comment:
We don't want to close application if our kafka library throws an error, the health checker for kafka producers and consumers will reconnect the application if we previously lost the connection.

Instead use the following snippet to log kafka errors:

```
...
	kafkaConsumer.Channels().LogErrors(ctx, "error received from kafka consumer, topic: "+<config topic>)
	kafkaProducer.Channels().LogErrors(ctx, "error received from kafka producer, topic: "+<config topic>)
	kafkaErrorProducer.Channels().LogErrors(ctx, "error received from kafka producer, topic: "+<config topic>)
...
```

This should exist before blocking switch statement